### ISSUE TITLE
Question!

### ISSUE THREAD

**Issue by dcflow** [jsetting32/Stripe-Connect-Parse#1]
Hello John,

I really like this and I have been searching for it for a long time, but I am having some troubles to implement it. 
I tried to deploy the folder WebServer to Heroku but I have to use build packs. 
Can you tell me some steps I need to follow to implement it in an iOS app?

Thank You!

**Comment by jsetting32** [comment]
So you basically want to implement the web server logic within the iOS app? I first tried to implement this but was unsuccessful as it isn't the recommended way to go about this. Typically you want a Web App (WebServer) to handle Stripe Connect authorization. When the user wants to link their Stripe Account to the application to start receiving payments from users, they click a button that redirects them to your web app, then the web app immediately directs the user to Stripe, once the user clicks the link in Stripe to link the account, they get redirected back to the URL that you (as the admin) setup in your account settings (which should be back to your web server). 

You then handle the response from Stripe connect, make a post back to stripe to get an access token for the user, then open your app by passing in a custom url scheme with the token you received from Stripe as a parameter and other things you feel is necessary to save to Parse.

Once your app opens up, you parse the parameters and save the values to the database under the current PFUser.

This probably doesn't answer your question but Stripe and other answers in StackOverflow recommend that you do not do all this through the iOS application, but through a web app (like django, or node, etc.)

**Comment by jsetting32** [comment]
I am currently trying to implement a production part of this application, which is a little frustrating as Stripe requires an 'https' redirect url, not 'http'. Also, the web client should not be talking to Django directly in Production. So I need to implement some middleware between the two. Still confused on how to do that but will update this repo with a FULL stack implementation on getting a production ready Stripe-Connect/Django/Parse/Cloud-Code/iOS Stack up and running

**Comment by dcflow** [comment]
Thank you very much for your help John! I saw some popular Ruby repositories that were for Stripe marketplaces. Do you think they could help you?

**Comment by jsetting32** [comment]
I kinda just wanted to get a full Django solution up and running. I have looked into making a Rails project but I want to be able to implement a full web app from the ground up just to get a better understanding of a full stack implementation. I have implemented a LAMP stack before but wanted to start working with web application frameworks, since most companies use them, at least from my understanding :P

**Comment by dcflow** [comment]
Ok I will be waiting for the next updates then. :)

**Comment by dcflow** [comment]
Hey John what about this module "HotSpot" .. where is this?

**Comment by jsetting32** [comment]
Oh thats just my iOS app name.

**Comment by dcflow** [comment]
Aaa ok .. the thing is that I get an error "No module with this name was found" (even if I replace it with my app name) when it reaches the server. Where is the point of connection between the app and the server, how does it understand that this is the name of the app?

**Comment by jsetting32** [comment]
So i think the problem here is that you need to setup the config/global.js file within the cloud code directory. I took out my keys of course. Just add yours and hopefully it works as expected. I think this is the issue.

But when you mean point of connection, do you mean when your iOS app asks to setup a Stripe account? Or just when running cloud code? 

Because if you are just running cloud code this is surely the issue.

**Comment by dcflow** [comment]
I haven't pushed Parse Cloud Code yet. 

I click Connect with Stripe and Safari opens up. I fill the form and when it gets redirected to my server I get the error "HotSpot module not found" because some python files have "from HotSpot import views" or "from HotSpot.models import User, Event". 

Changing HotSpot to my app name(Random) won't fix the issue. There must be a connection between the app and the server that I am missing.

![screen shot 2015-03-30 at 10 04 34 pm](https://cloud.githubusercontent.com/assets/6751025/6905532/e02f524e-d728-11e4-9e99-002c781d8723.png)

**Comment by jsetting32** [comment]
Hmm. Okay. Get rid of everything that references HotSpot within the views.py. Also, for the web server to open the app from safari, you need to implement a URL Scheme in your iOS app… I have mine as 'partyhard://'. You’ll notice this by viewing the bottom of the stripe_callback function in views.py.

Let me know if your still getting errors.

**Comment by dcflow** [comment]
I don' t think that will work .. there are many dependencies on HotSpot other than in views.py :
In urls.py  -> from HotSpot import views
In admin.py -> from HotSpot.models import User, Event
And the error first occurs at urls.py. 

Maybe there is some file missing. Do you have a settings.py anywhere?

**Comment by jsetting32** [comment]
Yes I do, but the only modifications I really made to the settings.py file was in the INSTALLED_APPS tuple, where I include hotspot. So I’m thinking the error is most likely coming from urls.py as well, like you said, since a lot references are made to ‘hotspot'. Since it needs the hotspot app urls.py file. 

In my honest opinion, I would just run a grep on the project directory for anything ‘hotspot’ and just change it to your application, or rather, just create a new app within your django project. Really the only thing you need the Web App to do is handle the redirects from Stripe, which is done within the views.py file. So copy and paste the functions within the views.py file into your app’s views.py file, setup the urls.py file to call the functions in views.py, and everything should work as expected.

I will upload the entire Django project in a bit so you can refer to it. I feel that once that is included, everything will be working just fine. Sorry for the inconvenience!

**Comment by jsetting32** [comment]
Hey so i updated the repo to include the entire django project rather than the single app. Go ahead and add your keys to the views.py project. Also, your going to have to setup your iOS app to 

[[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"http://127.0.0.1:8000/hotspotapi/stripe_auth"]];

Instead of [[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"http://127.0.0.1:8000/hotspot/stripe_auth"]];

Basically everything url related is now pointing from hotspot -> hotspotapi. Go ahead and change anything that says hotspotapi to your app name if you like, but I think you should just try and get it up and running, then experiment with different names. Please email me back if your still having issues

**Comment by dcflow** [comment]
Hey John sorry for the late answer,

I gave it a quick try and the oauth flow works perfect. Now I guess we have to implement all the other functions on the Stripe API(charging, transferring etc.) on the server. 

Thank you very much for your help!

**Comment by jsetting32** [comment]
Its no problem. I already implemented the methods in the parse cloud code, but if you want to add the functionality to the web server, that would be cool too! Less requests sent to Parse is great to have.

**Comment by dcflow** [comment]
Hey @jsetting32 how are you? Have you been working on any cool apps?

**Comment by jsetting32** [comment]
Hey @EMidee66. Sorry about the late response. Im doing fine. I just recently (5 months ago) got a job as an iOS software engineer and am just working on our product. I haven't been working on the side on anything new unfortunately. Work takes a lot out of me and i just like to relax when I get home. But Ill give you any update on when I do start a new project!
